### PR TITLE
[3.5] Added client keepalive arguments to the grpc-proxy

### PR DESCRIPTION
**Pull request by halleyshx**
[…m arguments to the grpc-proxy

Please read https://github.com/etcd-io/etcd/blob/main/CONTRIBUTING.md#contribution-flow.
](https://github.com/etcd-io/etcd/pull/17366#issue-2117590456)

### CODE REVIEW

**Comment by ahrtr** [comment]
This PR is backporting https://github.com/etcd-io/etcd/pull/17366

File: server/etcdmain/grpc_proxy.go [hunk 1]
```diff
@@ -58,7 +58,10 @@ var (
 	grpcProxyListenAddr                string
 	grpcProxyMetricsListenAddr         string
 	grpcProxyEndpoints                 []string
-	grpcProxyEndpointsAutoSyncInterval time.Duration
+  grpcProxyEndpointsAutoSyncInterval time.Duration
+	grpcProxyDialKeepAliveTime         time.Duration
```
Comment (ahrtr) [inline server/etcdmain/grpc_proxy.go:62]:
```suggestion
	grpcProxyEndpointsAutoSyncInterval time.Duration
	grpcProxyDialKeepAliveTime         time.Duration
```

**Comment by ahrtr** [comment]
Pls squash the commits

**Comment by halleyshx** [comment]
> Pls squash the commits

With a lot Merge, I failed. I make a new PR to replace it.>.<

**Pull request closed without merging**